### DISCUSSION TITLE
File upload hangs, unable to upload files.

### DISCUSSION THREAD

**Discussion by alexnederlof** [open-webui/open-webui#7898] [Issues]
## Installation Method

Kubernetes through provided helm chart

## Environment

- **Open WebUI Version:** 0.4.8
- **Operating System:** Linux
- **Browser (if applicable):** Chrome

Running behind Istio load balancer. 

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

Uploading any file by dragging it into a conversation works.

## Actual Behavior:

- Drag a file into the UI ( a simple hello.txt with contents "hello" for this example)
- File upload starts, UI spinner is shown
- I see `POST api/v1/files/` happen
- This hangs until the gateway times out. 

During that period, on the kubernetes pod we see:

```
open-webui-0 open-webui WARNI [python_multipart.multipart] Skipping data after last boundary
open-webui-0 open-webui INFO  [open_webui.apps.webui.routers.files] file.content_type: text/plain
open-webui-0 open-webui INFO  [open_webui.apps.retrieval.main] save_docs_to_vector_db: document hello.txt file-3c8907c2-5b0e-412e-b25d-6fda724df2e0
open-webui-0 open-webui Collection file-3c8907c2-5b0e-412e-b25d-6fda724df2e0 does not exist.
open-webui-0 open-webui INFO  [open_webui.apps.retrieval.main] adding to collection file-3c8907c2-5b0e-412e
![Uploading Screenshot 2024-12-16 at 21.01.31.png…]()
-b25d-6fda724df2e0

```

## Description

**Bug Summary:**
Seems like the server never responds, even though it correctly reads the file. If it's a PDF, I also see Tikka trigger and extract the contents correctly. 

## Reproduction Details

**Steps to Reproduce:**
[Outline the steps to reproduce the bug. Be as detailed as possible.]

## Logs and Screenshots

**Browser Console Logs:**
```
MessageInput.svelte:224 DragEvent {isTrusted: true, dataTransfer: DataTransfer, screenX: 1353, screenY: 707, clientX: 1103, …}
MessageInput.svelte:229 [File]
MessageInput.svelte:164 File {name: 'hello.txt', lastModified: 1734379218778, lastModifiedDate: Mon Dec 16 2024 21:00:18 GMT+0100 (Central European Standard Time), webkitRelativePath: '', size: 6, …} 'txt'
MessageInput.svelte:97 File {name: 'hello.txt', lastModified: 1734379218778, lastModifiedDate: Mon Dec 16 2024 21:00:18 GMT+0100 (Central European Standard Time), webkitRelativePath: '', size: 6, …}
FileItemModal.svelte:21 {type: 'file', file: '', id: null, url: '', name: 'hello.txt', …}
MessageInput.svelte:619 format PDF with pages and content.
Chat.svelte:837 submitPrompt What’s in this file 
index.ts:8 
        
        
       POST https://my.domain/api/v1/files/ 504 (Gateway Timeout)
window.fetch @ fetcher.js:76
c @ index.ts:8
k @ MessageInput.svelte:139
(anonymous) @ MessageInput.svelte:195
B @ MessageInput.svelte:163
oe @ MessageInput.svelte:230
index.ts:22 SyntaxError: Unexpected token 'u', "upstream r"... is not valid JSON
```

**Docker Container Logs:**

```
open-webui-0 open-webui WARNI [python_multipart.multipart] Skipping data after last boundary
open-webui-0 open-webui INFO  [open_webui.apps.webui.routers.files] file.content_type: text/plain
open-webui-0 open-webui INFO  [open_webui.apps.retrieval.main] save_docs_to_vector_db: document hello.txt file-3c8907c2-5b0e-412e-b25d-6fda724df2e0
open-webui-0 open-webui Collection file-3c8907c2-5b0e-412e-b25d-6fda724df2e0 does not exist.
open-webui-0 open-webui INFO  [open_webui.apps.retrieval.main] adding to collection file-3c8907c2-5b0e-412e
![Uploading Screenshot 2024-12-16 at 21.01.31.png…]()
-b25d-6fda724df2e0

```

**Screen shots**
<img width="816" alt="Screenshot 2024-12-16 at 21 09 20" src="https://github.com/user-attachments/assets/45c68afd-7733-4bb0-9dc8-751a3e256c96" />

**Comment by tjbck** [comment]
Likely something to do with your network config, keep us updated.

**Reply by alexnederlof** [reply]
every other call works correctly, there is no file limit nor any other limitation. What makes you think it's the network?

**Reply by tjbck** [reply]
Most people cannot reproduce this issue (including me, works as intended in all of the prod instances on my end), Keep us updated.

**Reply by alexnederlof** [reply]
Do you guys connect directly, or with a LB? Any experience with Istio being a problem or not?

**Comment by derek-assurity** [comment]
I've had the error message `index.ts:22 SyntaxError: Unexpected token 'u', "upstream r"... is not valid JSON` before. It is due to an upstream process returning an error message that is not JSON.

**Comment by alexnederlof** [comment]
Ruled out the load balancer. When I port forward to the pod directly, I see the same issue. The request completes before the gateway timeout, but it takes 1 minute and 22 second to upload a file of a few bytes

**Comment by alexnederlof** [comment]
OK ruled out the network too I think @tjbck

Same result after logging into the docker container, when I run this curl:

```
time curl 'http://localhost:8080/api/v1/files/' \
	 -H 'accept: application/json' \
	 -H 'accept-language: en-GB,en-US;q=0.9,en;q=0.8'  \ 
	 -H 'authorization: Bearer xxx'   \ 
	 -H 'cookie: intercom-device-id-nptsh54m=xx; intercom-id-nptsh54m=xx; istio="5fad212248fefb28"; token=xxx'   \ 
	 -H 'origin: https://my.domain'   \ 
	 -H 'priority: u=1, i'   \ 
	 -H 'referer: https://my.domain'   \ 
	 -H 'sec-ch-ua: "Google Chrome";v="131", "Chromium";v="131", "Not_A Brand";v="24"'   \ 
	 -H 'sec-ch-ua-mobile: ?0'   \ 
	 -H 'sec-ch-ua-platform: "macOS"'   \ 
	 -H 'sec-fetch-dest: empty'   \ 
	 -H 'sec-fetch-mode: cors'   \ 
	 -H 'sec-fetch-site: same-origin' \
	 -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36' \
   -F "file=@hello.text"
```

I get

```
{"id":"595587f4-ec6c-46ee-9b4b-402dc2a1d574","user_id":"c2e6e81f-6b41-4780-b6e1-54130c9d5883","hash":"ec98d3e57bb805eb0d724bf16c2fcd0d0dca3ddca1958ec30414d9f6861cd340","filename":"hello.text","data":{"content":"\n\n\n\n\n\n\n\n\ntext\n\n"},"meta":{"name":"hello.text","content_type":"application/octet-stream","size":5,"collection_name":"file-595587f4-ec6c-46ee-9b4b-402dc2a1d574"},"created_at":1734381878,"updated_at":1734381878}
real	1m0.561s
user	0m0.007s
sys	0m0.010s
```

It's suspicious that it takes one minute to load a file that's 5 characters right? that causes the gateway timeout

**Comment by spammenotinoz** [comment]
Might not be your issue, but I have experienced this issue numerous times, and was usually caused by connectivity issues to the embedding endpoint or an issue with the embedding endpoint itself.  If you change the embedding model engine, best to restart Open-WebUi.
I have been using OpenAI https://api.openai.com/v1, but that endpoint isn't without flaws.

For testing try using the Default Content Extraction Engine not Tika.

**Comment by alexnederlof** [comment]
Found the issue. I was running multiple instances of OpenWebui on a shared disk. I think there was a lock on the Chroma DB. WHen I switched to a dedicated chroma instance the issue went away.

I think the documentation could benefit from a description on how to run OpenWebUI with multiple instances. I'll see if I can contribute to that

Thanks for all your help!

**Reply by abiari** [reply]
@alexnederlof  thank you so much for this, I've been debugging for 2 days until i found this thread. Can you please share a guide or example on how to run a dedicated vector db for a multi instance deployment of Openwebui on a shared disk, I have the same use case as yours and want to shortcut the time i need to fix this issue. Thank you so much for this

**Comment by bgeneto** [comment]
Same here. I don't know how to debug... last messages are like that:

```
INFO  [open_webui.apps.retrieval.main] adding to collection file-d120eb20-d650-4ed3-ba38-8f64ce5515a0
Collection file-d120eb20-d650-4ed3-ba38-8f64ce5515a0 does not exist.
```
OR
```
WARNI [python_multipart.multipart] Skipping data after last boundary
Collection file-ea36acac-07a5-48bb-93d0-e93c13870546 does not exist.
```

**Comment by sao-coding** [comment]
synology DS220+
Unable to upload file
API timeout
![image](https://github.com/user-attachments/assets/867d9c3f-485d-42a4-bf68-1957b08934e6)
![image](https://github.com/user-attachments/assets/6b173fbd-7d70-46f6-abc4-69d2285a8914)

**Comment by BloodBlight** [comment]
I was seeing this issue while using Firefox, but using Chrome did not.  All local install here.

**Comment by regularRandom** [comment]
Same issue with 0.5.7. It doesn't allow neither send a query nor delete file, all other controls are unavailable, too. I can't even create a new chat.

**Comment by mariomaz87** [comment]
I have the same problem, the upload icon spins for a few minutes and then disappears. In the log I have only:

WARNI [python_multipart.multipart] Skipping data after last boundary
INFO  [open_webui.routers.files] file.content_type: text/csv
INFO  [open_webui.routers.retrieval] save_docs_to_vector_db: document history(1).csv file-1475452b-d17d-4b49-ad00-4a35cbde3243
INFO  [open_webui.routers.retrieval] adding to collection file-1475452b-d17d-4b49-ad00-4a35cbde3243

I'm using OpenAI as Embedding Model Engine, but the same happens with Default. I have tried Firefox and Chrome.

**Comment by netmb** [comment]
Same problem here:

WARNI [python_multipart.multipart] Skipping data after last boundary
INFO  [open_webui.routers.files] file.content_type: application/pdf
INFO  [open_webui.routers.retrieval] save_docs_to_vector_db: document Test.pdf file-aaa1a8d5-1805-4716-b747-2d475c6dbb77
INFO  [open_webui.routers.retrieval] adding to collection file-aaa1a8d5-1805-4716-b747-2d475c6dbb77
**Collection file-aaa1a8d5-1805-4716-b747-2d475c6dbb77 does not exist.**
INFO:     192.x.x.x:0 - "POST /api/v1/files/ HTTP/1.1" 200 OK

The upload in the frontend works, but no queries to the document are possible.

**Reply by netmb** [reply]
Problem solved for me. The message “Collection ...does not exist” seems to be normal. An older instance, where the upload still worked, gives the same error.
The problem was ultimately that a minimum score of 0.1 was entered in the document settings in the administrator area. This seems to be a problem in connection with the Reranker. If the minimum score is set to 0, everything works.

**Reply by mariomaz87** [reply]
> Problem solved for me. The message “Collection ...does not exist” seems to be normal. An older instance, where the upload still worked, gives the same error. The problem was ultimately that a minimum score of 0.1 was entered in the document settings in the administrator area. This seems to be a problem in connection with the Reranker. If the minimum score is set to 0, everything works.

I don’t have a minimum score setting…

**Comment by jcrabapple** [comment]
I am experiencing this same issue on v0.5.20.

**Reply by arunbugkiller** [reply]
I am experience the same issue, did you figure out the solution ?

**Comment by abiari** [comment]
If your OpenWebUI's instance is sitting behind a reverse proxy, try checking the proxy's timeout. We had the same issue where the proxy was time outing the connection before the vectorization and storing process could be finished, thus the "Collection does not exist" error message. We also had a very noticeable improvement in the speed of vectorization with openAI's embedder and pgvector as vector database.

**Reply by bar2098** [reply]
Thanks so much for this, totally helped out!

**Reply by abiari** [reply]
Glad it helped :)
